src/app/core/entity-components/previous-relations/previous-relations.component.ts

```diff
@@ -0,0 +1,133 @@
+import { Component, Input, OnInit } from "@angular/core";
+import { FormFieldConfig } from "../entity-form/entity-form/FormConfig";
+import moment from "moment";
+import { DynamicComponent } from "../../view/dynamic-components/dynamic-component.decorator";
+import { FontAwesomeModule } from "@fortawesome/angular-fontawesome";
+import { MatSlideToggleModule } from "@angular/material/slide-toggle";
+import { FormsModule } from "@angular/forms";
+import { MatTooltipModule } from "@angular/material/tooltip";
+import { NgIf } from "@angular/common";
+import { EntitySubrecordComponent } from "../entity-subrecord/entity-subrecord/entity-subrecord.component";
+import { PillComponent } from "../../common-components/pill/pill.component";
+import { ChildSchoolRelation } from "../../../child-dev-project/children/model/childSchoolRelation";
+import { Entity } from "../../entity/model/entity";
+import { RelatedEntitiesComponent } from "../entity-details/related-entities/related-entities.component";
+
+/**
+ * Display a list of entity subrecords (entities related to the current entity details view)
+ * which can be active or past/inactive.
+ *
+ * This component is similar to RelatedEntities but provides some additional UI to help users
+ * create a new entry if no currently active entry exists and
+ * show/hide inactive entries from the list.
+ *
+ * TODO: this is a more generic copy of ChildSchoolOverview - remove that once schema-generated indices are available
+ *  or should we just forget about that index for now? (longest running instance has total of 1000+ ChildSchoolRelation docs)
+ */
+@DynamicComponent("PreviousRelations")
```

Comment:
suggestions for a better name for this very welcome!

Reply:
The other very generic component is called `related-entities` so `related-time-period-entities` would be very consistent and make the difference/similarities clear.

Reply:
Especially as the entity has nothing to do with relations in the raw format. And also that is says previous is a assumption, use cases could also have elements with dates in the future.

Reply:
Yes. I always thought ChildSchoolRelation and ChildSchoolOverview is all about the m-n relation, turns out it is not and the key part is the time period thing :laughing: 

I guess `TimePeriod extends Entity` and `RelatedTimePeriodEntitiesComponent` would work quite well, as you suggest.

Reply:
or should it be `RelatedTimePeriodsComponent` ... :thinking: